.github/workflows/tag.yaml

```
@@ -0,0 +1,27 @@
+name: Tag Repo
+
+on:
+  workflow_dispatch:
+    inputs:
+      commit_sha:
+        description: 'Commit SHA'
+        required: true
+        default: 'HEAD'
+
+      release_number:
+        description: 'The number for this release e.g. v1.2.3'
+        required: true
+
+jobs:
+  tag:
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v4
+          with:
+            fetch-depth: 0
+        - name: Tag the repo
+          run: |
+            git config user.email "github-actions[bot]@users.noreply.github.com"
+            git config user.name "github-actions[bot]"
+            git tag -a ${{ github.event.inputs.release_number }} -m "Release ${{ github.event.inputs.release_number }}"
```

```suggestion
            git tag ${{ github.event.inputs.release_number }} --message="Release ${{ github.event.inputs.release_number }}"
```

I had to look up the `-a` since I hadn't used it. I'd remove it unless you found a reason it needs to be explicitly set. The docs say:

>  If -m <msg> or -F <file> is given and -a, -s, and -u <keyid> are absent, -a is implied.